### PR TITLE
Fix PHPCS report in checkout classes

### LINKED ISSUE #8466: Checkout: fix WordPress.Security phpcs reports
Part of https://github.com/Automattic/woocommerce-payments/issues/8434

Check and fix reports in the following files:

* `includes/express-checkout/class-wc-payments-express-checkout-button-display-handler.php`
* `includes/class-wc-payments-checkout.php`
* `includes/class-wc-payments-payment-request-button-handler.php`

### PR DESCRIPTION
**Pull request by timur27**
Fixes https://github.com/Automattic/woocommerce-payments/issues/8466

> Note: The command to run which will check files listed in https://github.com/Automattic/woocommerce-payments/issues/8466
```text
./vendor/bin/phpcs --standard=phpcs.xml.dist --sniffs='WordPress.Security.EscapeOutput,WordPress.Security.ValidatedSanitizedInput' includes/express-checkout/class-wc-payments-express-checkout-button-display-handler.php includes/class-wc-payments-checkout.php includes/class-wc-payments-payment-request-button-handler.php
```

#### Changes proposed in this Pull Request
This PR
* adds `isset` checks when getting the request param in `class-wc-payments-express-checkout-button-display-handler.php`
* mutes the phpcs report for non-escaped output in `class-wc-payments-checkout.php` when using `WC_Payments_Utils::esc_interpolated_html()` which itself escapes the return value 
* escapes the error message in `class-wc-payments-payment-request-button-handler.php`
* explicitly casts the product ID injected into the error message string

#### Testing instructions
Since the changes with`phpcs` reports were reverted, we need to work this around a little bit.

* `git fetch origin bc8ae14f29921f43f105a2240d21fe5c23ee932a:bc8ae14f29921f43f105a2240d21fe5c23ee932a`
* `git checkout bc8ae14f29921f43f105a2240d21fe5c23ee932a`
* run the command provided above and confirm that you see the errors for every file (5 errors overall)
* `git merge fix/phpcs-reports-checkout`
* run the command once more and no errors should appear

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [ ] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

File: includes/class-wc-payments-payment-request-button-handler.php [hunk 1]
```diff
@@ -251,12 +251,13 @@ public function get_product_price( $product, ?bool $is_deposit = null, int $depo
...
+                __( 'Express checkout does not support products without prices! Please add a price to product #%d', 'woocommerce-payments' ),
+                (int) $product->get_id()
+            );
 			throw new Invalid_Price_Exception(
-				sprintf(
-					// Translators: %d is the numeric ID of the product without a price.
-					__( 'Express checkout does not support products without prices! Please add a price to product #%d', 'woocommerce-payments' ),
-					$product->get_id()
-				)
+				esc_html( $error_message )
```
Comment (timur27) [inline includes/class-wc-payments-payment-request-button-handler.php]: Two things added here: 
1. escaping the prepared error message
2. explicitly cast product ID to `int` type (this one is IMO not necessarily but can be a good additional measure)

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `8475` or branch name `fix/phpcs-reports-checkout` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&woocommerce-payments-dev-tools&branches.woocommerce-payments=fix/phpcs-reports-checkout) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: 7fce771b559a1ebb54d6feefbf936190839bab33
- Build time: 2024-03-26 11:24:58 UTC

_Note: the build is updated when a new commit is pushed to this PR._

File: includes/express-checkout/class-wc-payments-express-checkout-button-display-handler.php [hunk 2]
```diff
@@ -173,8 +173,8 @@ function( $js_config ) use ( $order ) {
 					$user_email = isset( $_POST['email'] ) ? sanitize_email( wp_unslash( filter_input( INPUT_POST, 'email' ) ) ) : $session_email;
 
 					$js_config['order_id']      = $order->get_id();
-					$js_config['pay_for_order'] = sanitize_text_field( wp_unslash( $_GET['pay_for_order'] ) );
-					$js_config['key']           = sanitize_text_field( wp_unslash( $_GET['key'] ) );
+					$js_config['pay_for_order'] = sanitize_text_field( wp_unslash( isset( $_GET['pay_for_order'] ) ) );
+					$js_config['key']           = sanitize_text_field( wp_unslash( isset( $_GET['key'] ) ) );
```
Comment (frosso) [inline includes/express-checkout/class-wc-payments-express-checkout-button-display-handler.php]: `isset` returns a boolean. With the new changes, is this going to return the expected output?

Comment (timur27) [reply includes/express-checkout/class-wc-payments-express-checkout-button-display-handler.php]: Yes, good point. Actually, I see that `isset` is checked for both params above in the beginning of the function so I'll just go ahead and add `phpcs:ignore`

Comment (frosso) [reply includes/express-checkout/class-wc-payments-express-checkout-button-display-handler.php]: Sounds good, I agree 👍

**Review by frosso — APPROVED** [review]
To the moooooon! 🚀

**Pull request merged**